src/Infer/Solve.hs

```diff
@@ -494,9 +524,9 @@ inferWhere env whereExp@(Meta _ loc (Src.Where exp iss)) = do
       (Src.PVar    v     , t'  ) -> return $ extendVars e (v, Forall [] t')
 
       (Src.PRecord fields, tipe) -> do
```

Comment:
What is `tipe` here?

Reply:
A type, but type is a reserved word in Haskell. I saw that in the code of Elm and in most places I refer to types simply as "t" or "t'", or somethingType. I think the whole module should be refactored to name bindings that contain values of type Type in a consistent manner.